.circleci/workflows.yml

```
@@ -0,0 +1,1198 @@
+            # this task is overwriting the content produced by generate-sql;
+            # the behaviour here is additive, generated DAGs are just added to
+            # the generated-sql output
+            - persist_to_workspace:
+                root: /tmp/workspace
+                paths:
+                  - generated-sql
+      - unless:
+          condition: *validate-sql
+          steps:
+            - *skip
+  validate-dags:
+    executor:
+      name: python/default
+      tag: 3.10.12
+    steps:
+      - when:
+          condition: *validate-sql
+          steps:
+            - checkout
+            - run:
+                name: Early return when job not modified
+                command: |
+                  if [ "$CIRCLE_BRANCH" = main ]; then
+                    echo "Run job because branch is $CIRCLE_BRANCH"
+                  elif git log --format=%B --no-merges -n 1 |
+                      grep -qF '[run-tests]'; then
+                    echo "Run job because [run-tests] in commit message"
+                  elif ! git diff --quiet origin/main... \
+                      -- "$(git rev-parse --show-toplevel)"/{.circleci,dags}; then
+                    echo "Run job because .circleci/ and/or dags/ were modified" \
+                      "since branching off main"
+                  else
+                    echo "Skipping job because .circleci/ and dags/ were not modified"
+                    circleci-agent step halt
+                  fi
+            - run:
+                name: Pull telemetry-airflow
+                command: |
+                  git clone https://github.com/mozilla/telemetry-airflow.git ~/telemetry-airflow
+            - *attach_generated_sql
+            - *copy_generated_sql
+            - run:
+                name: Replace telemetry-airflow DAGs with BigQuery ETL DAGs
+                command: |
+                  rm ~/telemetry-airflow/dags/* -f || true
+                  cp -a /tmp/workspace/generated-sql/dags/. ~/telemetry-airflow/dags/
+            - run:
+                name: Install telemetry-airflow dependencies
+                command: |
+                  cd ~/telemetry-airflow
+                  virtualenv .venv
+                  source .venv/bin/activate
+                  pip install -r requirements.txt
+            - run:
+                name: 🧪 Test valid DAGs
+                command: |
+                  cd ~/telemetry-airflow
+                  source .venv/bin/activate
+                  python -m pytest tests/dags/test_dag_validity.py --junitxml=~/telemetry-airflow/test-results/junit.xml
+            - store_test_results:
+                path: ~/telemetry-airflow/test-results/junit.xml
+      - unless:
+          condition: *validate-sql
+          steps:
+            - *skip
+  test-routines:
+    docker: *docker
+    steps:
+      - when:
+          condition: &validate-routines
+            or:
+              - << pipeline.parameters.validate-routines >>
```

Should this also check the `deploy` parameter like the other validate conditions?
```suggestion
          condition: &validate-routines
            or:
              - << pipeline.parameters.validate-routines >>
              - << pipeline.parameters.deploy >>
```

---

yes https://github.com/mozilla/bigquery-etl/pull/5115